### ISSUE TITLE
Fix automation

### ISSUE THREAD

**Issue by jonsterling** [RedPRL/sml-redprl#569]
We need to fix `auto` soon; @cangiuli and I believe strongly that this should be our principle priority at this time, and it needs to happen before we write much more RedPRL code, and before we can really start recommending that people try RedPRL.

A big source of frustration with `auto` is the small-step nature of it; in order to implement the needed proof search tactics, we need a ton of backtracking (which can be avoided in a big-step version). Instead, we should engineer things so that problems are posed in the order that they can be solved, and that they are solved in one big step, modulo emitting subgoals that require user intervention.

Related is #438; I open this as a separate ticket because I want to track the task of fixing `auto` *regardless* of the specificities of that proposal.

**Comment by favonia** [comment]
What exactly needs to be fixed now?

**Comment by jonsterling** [comment]
I think it’s better than it was, but the main problem remains that it is a little unpredictable, often loops, and we don’t really understand how to study the proof search scientifically.
